### PR TITLE
[plugin] - content pagenavigation SQL error  when order by author

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #16127 .

### Summary of Changes
added join when order by field `u.name`

### Testing Instructions
https://github.com/joomla/joomla-cms/issues/16127#issue-229973692
 + content navigation work as before 

### Expected result
no error

### Actual result
sql error

### CODE REVIEW

**Comment by franz-wohlkoenig** [comment]
will test Today.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 2c50de41fb7e2e2c2193fe075ff92a0dcf17f278<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16556">issues.joomla.org/tracker/joomla-cms/16556</a>.</sub>

**Comment by andrepereiradasilva** [comment]
hi @alikon but that means you will not order by Author (created by alias many times can be empty).
image

I think the right solution is: if ordering by author, join the users table also.

**Comment by brianteeman** [comment]
This change is not correct. It will only work if an alias is set. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16556">issues.joomla.org/tracker/joomla-cms/16556</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on 2c50de41fb7e2e2c2193fe075ff92a0dcf17f278<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16556">issues.joomla.org/tracker/joomla-cms/16556</a>.</sub>

**Comment by richard67** [comment]
I agree with @andrepereiradasilva : If ordering by author, join the users table also.

**Comment by alikon** [comment]
oops still in holiday :blush: , i'll add the join  with the users table when needed

File: plugins/content/pagenavigation/pagenavigation.php [hunk 1]
```diff
@@ -137,7 +137,12 @@ public function onContentBeforeDisplay($context, &$row, &$params, $page = 0)
 			$query->select('a.id, a.title, a.catid, a.language,' . $case_when . ',' . $case_when1)
 				->from('#__content AS a')
 				->join('LEFT', '#__categories AS cc ON cc.id = a.catid');
-			$query->join('LEFT', '#__users AS u ON u.id = a.created_by');
+
+			if (($order_method === 'author') || ($order_method === 'rauthor'))
```
Comment (Quy) [inline plugins/content/pagenavigation/pagenavigation.php]: Remove unnecessary parentheses.

`if ($order_method === 'author' || $order_method === 'rauthor')`

**Comment by alikon** [comment]
ready to test

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 6be3a60a6500cf430ee0770c41cf3f5896fa95b8<br /><br />Here I was getting this error before patch:

```
[14-Jun-2017 06:49:43 UTC] PHP Notice:  Undefined variable: previousExceptionHandler in /Applications/MAMP/htdocs/testwindows/trunkgitnew/plugins/system/redirect/redirect.php on line 112
[14-Jun-2017 06:49:43 UTC] PHP Stack trace:
[14-Jun-2017 06:49:43 UTC] PHP   1. PlgSystemRedirect::handleException() /Applications/MAMP/htdocs/testwindows/trunkgitnew/plugins/system/redirect/redirect.php:0
[14-Jun-2017 06:49:43 UTC] PHP   2. PlgSystemRedirect::doErrorHandling() /Applications/MAMP/htdocs/testwindows/trunkgitnew/plugins/system/redirect/redirect.php:91
[14-Jun-2017 06:49:43 UTC] PHP Fatal error:  Function name must be a string in /Applications/MAMP/htdocs/testwindows/trunkgitnew/plugins/system/redirect/redirect.php on line 112
[14-Jun-2017 06:49:43 UTC] PHP Stack trace:
[14-Jun-2017 06:49:43 UTC] PHP   1. PlgSystemRedirect::handleException() /Applications/MAMP/htdocs/testwindows/trunkgitnew/plugins/system/redirect/redirect.php:0
[14-Jun-2017 06:49:43 UTC] PHP   2. PlgSystemRedirect::doErrorHandling() /Applications/MAMP/htdocs/testwindows/trunkgitnew/plugins/system/redirect/redirect.php:91

```

This PR solves it.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16556">issues.joomla.org/tracker/joomla-cms/16556</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 6be3a60a6500cf430ee0770c41cf3f5896fa95b8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16556">issues.joomla.org/tracker/joomla-cms/16556</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Pull request merged**